main.go

```diff
@@ -144,7 +147,11 @@ func main() {
 			}
 
 			action := pr.GetAction()
-			parsePullRequest(action, pr, &conf)
+
+			// parse one pull request at a time since we are running some git commands
+			conf.lock.Lock()
```

Comment:
If you want to serialize the requests why not spawn a goroutine in main and push parsed build config through a channel?

Since `parsePullRequest()` should not modify anything, you could change it to `parsePullRequest(action string, pr ..) []buildConfig` (since a PR may need more builds), then have a channel `chan buildConfig`;

```
...
builds, err := parsePullRequest(config, action, pr)
// error handling
for _, build := range builds {
    bc -> config.builds
}
...
```

In builds goroutine

```
func doBuilds() {
...
for {
    var build := <- conf.builds
    if err := triggerBuild(conf, buildConf); err != nil {
        log.Warnf("... error")
    } else {
        log.Info("build started")
    }
}
```